book.rb

```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rental
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
+end
```

Comment:
- [ ] Kindly create a custom method called `add_rental` that takes a `date` and `person` as parameters. In the method, you should return a new instance of the `Rental` class. Remember that the `Book` and the `Person` object have a many-to-many relationship through the `Rental` class. This means that many people can rent a book on different dates and a single person can rent different books on different dates also. 

#### Hint:
```ruby
def add_rental(person, date)
    Rental.new(date, self, person)
end
```

If you remember, in the initialize method of the `Rental` class, both the book and person objects that are passed in as parameters are pushing the instance of the rental into their array attributes at creation:

```ruby
# these lines of codes
book.rental << self
person.rental << self
```
This way, each book object can keep track of all the people that rented it while each person object can keep track of all the books that were rented by that same person through their rental array attributes. 👍

Reply:
👍🏽